code/modules/ruins/spaceruin_code/hilbertshotel.dm

```diff
@@ -334,6 +369,16 @@ GLOBAL_VAR_INIT(hhMysteryRoomNumber, rand(1, 999999))
 	if(prob(0.135685)) //Because screw you
 		qdel(H)
 		return
+
+	// Prepare for...
+	var/mob/unforeseen_consequences = get_atom_on_turf(H, /mob)
+
+	// Turns out giving anyone who grabs a Hilbert's Hotel a free, complementary warp whistle is probably bad.
+	// Let's gib the last person to have selected a room number in it.
+	if(unforeseen_consequences)
+		to_chat(unforeseen_consequences, "<span class='warning'>\The [H] starts to resonate. Forcing it to enter itself induces a bluespace paradox, violently tearing your body apart.</span>")
+		unforeseen_consequences.gib()
```

Comment:
I dont fan of this, But why not first move `unforeseen_consequences` to targetturf (place where hilbertshotel land) before gibbing.

Reply:
That's the risk to using Hilbert's Hotel.

It's an item you can use to teleport away to some unknown location that nobody else can access without knowing the specific number you used between 1 and 16777216. You can then look through the peep hole as well as exit at any time you want.

Better be careful with how you use this power. Here's hoping you told someone what hotel room you were going to!

Reply:
A say about, move user to safeturf and next gib. to leave gibs on station level. not in hilbert hotel room. 
Or you want leave gibs in room?

Reply:
Gibbing them in the hotel room is the intent, yes. I hope they told their friends which room they're using if they try it.

Reply:
If you gib them in the hotel room anything they're carrying will be trapped there too. That could mean rare or unique items or traitor objectives being lost. Wouldn't it make more sense to gib them somewhere where these could conceivably be reached regardless of whether they've told people the room number (which they probably haven't).